### PR TITLE
feat: update max_persisted_sequence_number in table on persist

### LINKED ISSUE #3850: Ingester: persist must update table_data max sequence
Once PR #3849 lands, the ingester persistence should be updated so that when it is done persisting a parquet file, the `parquet_max_sequence_number` in `TableData` should be changed when `clear_persisting` is called.

### PR DESCRIPTION
**Pull request by pauldix**
This includes a bit of a refactor in the locking structure of the buffer data. Locking at the partition collection and within the partition data was making things more complex than they needed to be. The partitions in the buffer are only there temporarily until they get persisted. Locking on the table simplifies things a bit and makes it more clear when the table state is being modified since it no longer has any interior mutability. Having access to separate partitions without the same lock isn't something we need because queries will hit all partitions and data is brought in sequentially, regardless of which partition it is hitting in a sequencer.

Closes #3850

### CODE REVIEW

File: ingester/src/data.rs [hunk 1]
```diff
@@ -616,89 +648,78 @@ impl TableData {
...
     }
 
     /// Snapshot whatever is in the buffer and return a new vec of the
     /// arc cloned snapshots
-    pub fn snapshot(&self) -> Result<Vec<Arc<SnapshotBatch>>> {
-        let mut data = self.inner.write();
-        data.snapshot().context(SnapshotSnafu)?;
-        Ok(data.snapshots.to_vec())
+    pub fn snapshot(&mut self) -> Result<Vec<Arc<SnapshotBatch>>> {
+        println!("snapshot on PartitionData");
```
Comment (NGA-TRAN) [inline ingester/src/data.rs:716]: Is this a leftover or kept on purpose?

File: ingester/src/data.rs [hunk 2]
```diff
@@ -858,6 +872,7 @@ impl DataBuffer {
     /// Move `BufferBatch`es to a `SnapshotBatch`.
     pub fn snapshot(&mut self) -> Result<(), mutable_batch::Error> {
         if !self.buffer.is_empty() {
+            println!("buffer not empty");
```
Comment (NGA-TRAN) [inline ingester/src/data.rs:875]: a lot of println here too

**Review by NGA-TRAN — APPROVED** [review]
Looks reasonable to me. Better structure and locking mechanism I think.
This PR will conflict with the PR #3825. I hope you will let #3825 merged first 😄  CC @carols10cents

**Comment by pauldix** [comment]
@NGA-TRAN ok this is ready for another look. It ended up being a bit more ugly because the `test_util.rs` code ends up using things that I wanted to make private. I'll have to do a refctor later that removes test_util and instead has the setup things inside of `data.rs`. I'd prefer to have `TableData` and `PartitionData` be private implementation details of `data.rs`. I think ideally the only thing that should leak from the `NamespaceData` are the `Snapshot` and `QueryableBatch`.

**Comment by NGA-TRAN** [comment]
@pauldix This looks good. Yeah, the test_util.rs was created to make tests reusable in many modules but as a consequence, we need `pub(crate)` for many structs and functions

**Pull request merged**